src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -454,7 +468,23 @@ protected virtual async Task<ExecutionResult> ExecuteRequestAsync(HttpContext co
     /// <see cref="WriteJsonResponseAsync{TResult}(HttpContext, HttpStatusCode, TResult)"/>.
     /// </summary>
     protected virtual string SelectResponseContentType(HttpContext context)
-        => CONTENTTYPE_GRAPHQLJSON;
+    {
+        var acceptHeaders = context.Request.GetTypedHeaders().Accept;
+        if (acceptHeaders != null)
+        {
+            for (int i = 0; i < acceptHeaders.Count; i++)
+            {
+                for (int j = 0; j < _validMediaTypes.Length; j++)
+                {
+                    var (mediaType, responseType) = _validMediaTypes[j];
+                    if (acceptHeaders[i].MediaType.Equals(mediaType, StringComparison.OrdinalIgnoreCase))
+                        return responseType;
+                }
+            }
+        }
+
+        return CONTENTTYPE_GRAPHQLRESPONSEJSON;
```

Comment:
The problem here is that if you only have a single misbehaving client, then you would need to subclass the `GraphQLHttpMiddleware` and recreate the logic above with a different fall-back. It would still be really good to be able to _just_ override the default response content type.

Reply:
I agree.